src/java.base/share/classes/sun/security/pkcs/PKCS7.java

```diff
@@ -844,65 +844,6 @@ private static byte[] constructToken(byte[] signature,
         return p7out.toByteArray();
     }
 
-    /**
-     * Assembles a PKCS #7 signed data message that optionally includes a
-     * signature timestamp.
-     *
-     * @param signature the signature bytes
-     * @param signerChain the signer's X.509 certificate chain
-     * @param content the content that is signed; specify null to not include
-     *        it in the PKCS7 data
-     * @param signatureAlgorithm the name of the signature algorithm
-     * @param tsaURI the URI of the Timestamping Authority; or null if no
-     *         timestamp is requested
-     * @param tSAPolicyID the TSAPolicyID of the Timestamping Authority as a
-     *         numerical object identifier; or null if we leave the TSA server
-     *         to choose one. This argument is only used when tsaURI is provided
-     * @return the bytes of the encoded PKCS #7 signed data message
-     * @throws NoSuchAlgorithmException The exception is thrown if the signature
-     *         algorithm is unrecognised.
-     * @throws CertificateException The exception is thrown if an error occurs
-     *         while processing the signer's certificate or the TSA's
-     *         certificate.
-     * @throws IOException The exception is thrown if an error occurs while
-     *         generating the signature timestamp or while generating the signed
-     *         data message.
-     */
-    @Deprecated(since="16", forRemoval=true)
-    public static byte[] generateSignedData(byte[] signature,
```

Comment:
Are you sure you want to remove this as part of this change? If there was a reason this was deprecated for removal (perhaps some apps did use it even though they should not), then before removing it it probably makes sense to include it in a CSR, and maybe the RN too.

Reply:
Ok, I see. It probably should have been removed as part of https://bugs.openjdk.org/browse/JDK-8303410. So I'm ok with the cleanup.

It probably makes sense to rename `newGenerateSignedData` to `generateSignedData` (since there is only one method now).